modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java

```diff
@@ -662,8 +662,15 @@ private boolean _hasFriendlyURLEntryWithUrlTitle(
 			friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
 				groupId, classNameId, languageId, urlTitle);
 
-		if ((friendlyURLEntryLocalization != null) &&
-			(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {
+		FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
+			friendlyURLEntryLocalizationPersistence.fetchByG_C_NotL_U_First(
```

Comment:
What do you think on put the code this way

```
FriendlyURLEntryLocalization friendlyURLEntryLocalization =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
		groupId, classNameId, languageId, urlTitle);

if ((friendlyURLEntryLocalization != null) &&
	(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {

	return true;
}

FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_NotL_U_First(
		groupId, classNameId, languageId, urlTitle, null);

if ((friendlyURLEntryLocalizationNotLang != null) &&
	(friendlyURLEntryLocalizationNotLang.getClassPK() != notClassPK)) {

	return true;
}
```


double return vs a call to DB (maybe) not needed.